### PR TITLE
Fix playlists

### LINKED ISSUE #12892: Items in playlists cannot be moved. The item goes back in the interface back to its previous location and an error "Index was outside the bounds of the array" appears in the logs.
### This issue respects the following points:

- [X] This is a **bug**, not a question or a configuration issue; Please visit our forum or chat rooms first to troubleshoot with volunteers, before creating a report. The links can be found [here](https://jellyfin.org/contact/).
- [X] This issue is **not** already reported on [GitHub](https://github.com/jellyfin/jellyfin/issues?q=is%3Aopen+is%3Aissue) _(I've searched it)_.
- [X] I'm using an up to date version of Jellyfin Server stable, unstable or master; We generally do not support previous older versions. If possible, please update to the latest version before opening an issue.
- [X] I agree to follow Jellyfin's [Code of Conduct](https://jellyfin.org/docs/general/community-standards.html#code-of-conduct).
- [X] This report addresses only a single issue; If you encounter multiple issues, kindly create separate reports for each one.

### Description of the bug

After upgrading to 10.10.0, playlist items can not be moved. (Downgrading back to 10.9.11 solves the issues, so it seems likely due to the upgrade). I have previously had a similar issue where a playlist containing a removed item could no longer be rearranged or edited (that seemed like it was fixed). I tried creating new playlists, restarting the docker container, restarting the docker service and restarting the system. I also tried connecting through the local ip and http://  rather than through a nginx-proxy-manager and https://. Nothing solved the issue until I reverted the docker container to the 10.9.11 image.

Since part of the error in the logs said "Index was outside the bounds of the array," I tried moving an item to different places but all locations in a playlist led to the same error.

### Reproduction steps

1. Create a playlist and add items.
2. Rearrange the items in the jellyfin interface.

### What is the current _bug_ behavior?

The playlist order does not change and an error appears in the logs saying " [ERR] [31] Jellyfin.Api.Middleware.ExceptionMiddleware: Error processing request. URL POST /Playlists/c58b999842431004eb431fecfc2037f2/Items/be6a6882828d47dfba59eaa928003ca1/Move/0.
System.IndexOutOfRangeException: Index was outside the bounds of the array."

### What is the expected _correct_ behavior?

The playlist order changes.

### Jellyfin Server version

10.10.0+

### Specify commit id

_No response_

### Specify unstable release number

_No response_

### Specify version number

_No response_

### Specify the build version

10.10.0

### Environment

```markdown
- OS: Debian Bullseye
- Linux Kernel: 6.1.0-25
- Virtualization: Docker
- Clients: Browser, android, iOS
- Browser: Chrome
- FFmpeg Version:
- Playback Method: Direct Play
- Hardware Acceleration: none
- GPU Model:
- Plugins:
- Reverse Proxy: nginx-proxy-manager
- Base URL:
- Networking: 
- Storage: local
```

### Jellyfin logs

```shell
[17:31:46] [ERR] [31] Jellyfin.Api.Middleware.ExceptionMiddleware: Error processing request. URL POST /Playlists/c58b999842431004eb431fecfc2037f2/Items/be6a6882828d47dfba59eaa928003ca1/Move/0.
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Emby.Server.Implementations.Playlists.PlaylistManager.MoveItemAsync(String playlistId, String entryId, Int32 newIndex)
   at Jellyfin.Api.Controllers.PlaylistsController.MoveItem(String playlistId, String itemId, Int32 newIndex)
   at lambda_method1267(Closure, Object)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Jellyfin.Api.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager)
   at Jellyfin.Api.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager)
   at Jellyfin.Api.Middleware.IPBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager)
   at Jellyfin.Api.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Api.Middleware.QueryStringDecodingMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Api.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext)
   at Jellyfin.Api.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)
   at Jellyfin.Api.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context, IServerConfigurationManager serverConfigurationManager)
   at Jellyfin.Api.Middleware.ExceptionMiddleware.Invoke(HttpContext context)
```

### FFmpeg logs

_No response_

### Client / Browser logs

_No response_

### Relevant screenshots or videos

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Shadowghost**
There is for sure a cleaner solution than this one.

**Issue:** For some reason the `Id` of `LinkedItem` is different every time they are read.

**Changes**
* Use `LinkedItem.ItemId` instead of `LinkedItem.Id` since the latter seems to be non-static
* Remove `LinkedItem.Id` alltogether
* Calculate new Index on moved item based on accessible items
* Fully remove option for duplicate playlist entries and cleanup duplicates from playlists

**Issues**
Fixes #12892

### CODE REVIEW

File: MediaBrowser.Controller/Extensions/ConfigurationExtensions.cs [hunk 1]
```diff
@@ -120,14 +120,6 @@ public static bool GetFFmpegSkipValidation(this IConfiguration configuration)
         public static bool GetFFmpegImgExtractPerfTradeoff(this IConfiguration configuration)
             => configuration.GetValue<bool>(FfmpegImgExtractPerfTradeoffKey);
 
-        /// <summary>
-        /// Gets a value indicating whether playlists should allow duplicate entries from the <see cref="IConfiguration"/>.
-        /// </summary>
-        /// <param name="configuration">The configuration to read the setting from.</param>
-        /// <returns>True if playlists should allow duplicates, otherwise false.</returns>
-        public static bool DoPlaylistsAllowDuplicates(this IConfiguration configuration)
-            => configuration.GetValue<bool>(PlaylistsAllowDuplicatesKey);
```
Comment (nielsvanvelzen) [inline MediaBrowser.Controller/Extensions/ConfigurationExtensions.cs:129]: Doesn't `PlaylistsAllowDuplicatesKey` need to be removed as well?

Comment (Shadowghost) [reply MediaBrowser.Controller/Extensions/ConfigurationExtensions.cs:129]: Yes

Comment (nielsvanvelzen) [reply MediaBrowser.Controller/Extensions/ConfigurationExtensions.cs:129]: Now the build is failing, I think you need to remove some more stuff?

**Review by crobibero — APPROVED** [review]

**Review by JPVenson — APPROVED** [review]

**Comment by pixel24** [comment]
Is there any estimate on when this bug might be fixed? I work on my playlists daily, and I only noticed the issue after already adding quite a bit to my playlists, which I’d lose if I had to roll back.

**Comment by talkingseedling** [comment]
After upgrading to 10.10.2, Jellyfin wouldn't come up. I think it's related to this PR. Error below. I fixed by rolling back to 10.10.1 and deleting my playlist folder I no longer use. The music folders are mounted in docker as **read only**, which may have caused this issue.

[20:07:33] [ERR] [1] Jellyfin.Server.Migrations.MigrationRunner: Could not apply migration 'RemoveDuplicatePlaylistChildren'
System.IO.DirectoryNotFoundException: Could not find a part of the path '/data/music/music/Playlists/Triple J 1999.wpl'.
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirError)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode, Boolean failForSymlink, Boolean& wasSymlink, Func`4 createOpenException)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, UnixFileMode openPermissions, Int64& fileLength, UnixFileMode& filePermissions, Boolean failForSymlink, Boolean& wasSymlink, Func`4 createOpenException)
   at System.IO.File.OpenHandle(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.File.WriteToFile(String path, FileMode mode, String contents, Encoding encoding)
   at Emby.Server.Implementations.Playlists.PlaylistManager.SavePlaylistFile(Playlist item)
   at Jellyfin.Server.Migrations.Routines.RemoveDuplicatePlaylistChildren.Perform()
   at Jellyfin.Server.Migrations.MigrationRunner.PerformMigrations(IMigrationRoutine[] migrations, MigrationOptions migrationOptions, Action`1 saveConfiguration, ILogger logger)
[20:07:33] [FTL] [1] Main: Error while starting server
System.IO.DirectoryNotFoundException: Could not find a part of the path '/data/music/music/Playlists/Triple J 1999.wpl'.
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirError)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode, Boolean failForSymlink, Boolean& wasSymlink, Func`4 createOpenException)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, UnixFileMode openPermissions, Int64& fileLength, UnixFileMode& filePermissions, Boolean failForSymlink, Boolean& wasSymlink, Func`4 createOpenException)
   at System.IO.File.OpenHandle(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.File.WriteToFile(String path, FileMode mode, String contents, Encoding encoding)
   at Emby.Server.Implementations.Playlists.PlaylistManager.SavePlaylistFile(Playlist item)
   at Jellyfin.Server.Migrations.Routines.RemoveDuplicatePlaylistChildren.Perform()
   at Jellyfin.Server.Migrations.MigrationRunner.PerformMigrations(IMigrationRoutine[] migrations, MigrationOptions migrationOptions, Action`1 saveConfiguration, ILogger logger)
   at Jellyfin.Server.Migrations.MigrationRunner.Run(CoreAppHost host, ILoggerFactory loggerFactory)
   at Jellyfin.Server.Program.StartServer(IServerApplicationPaths appPaths, StartupOptions options, IConfiguration startupConfig)

**Pull request merged**